packages/frontend/admin/src/modules/accounts/components/ceate-user-panel.tsx

```diff
@@ -0,0 +1,189 @@
+import { Button } from '@affine/admin/components/ui/button';
+import { Input } from '@affine/admin/components/ui/input';
+import { Label } from '@affine/admin/components/ui/label';
+import { Separator } from '@affine/admin/components/ui/separator';
+import { Switch } from '@affine/admin/components/ui/switch';
+import { useAsyncCallback } from '@affine/core/hooks/affine-async-hooks';
+import {
+  useMutateQueryResource,
+  useMutation,
+} from '@affine/core/hooks/use-mutation';
+import {
+  addToAdminMutation,
+  addToEarlyAccessMutation,
+  createUserMutation,
+  EarlyAccessType,
+  FeatureType,
+  listUsersQuery,
+} from '@affine/graphql';
+import { CheckIcon, XIcon } from 'lucide-react';
+import { useCallback, useMemo, useState } from 'react';
+import { toast } from 'sonner';
+
+import { useRightPanel } from '../../layout';
+
+export function CreateUserPanel() {
+  const { closePanel } = useRightPanel();
+  const [name, setName] = useState('');
+  const [password, setPassword] = useState('');
+  const [email, setEmail] = useState('');
+  const [features, setFeatures] = useState<FeatureType[]>([]);
+
+  const disableSave = useMemo(() => !name || !email, [name, email]);
+
+  const { trigger: createUser } = useMutation({
```

Comment:
we might need to involve the framework system as `@affine/core` used

Reply:
Can I understand it as abstracting this asynchronous logic into React hooks?

Reply:
yeah